ReactiveCocoa/Swift/Lifetime.swift

```diff
@@ -0,0 +1,36 @@
+import enum Result.NoError
+
+/// Provides a signal that completes when the lifetime object is deinitialized.
+///
+/// When assigned to a property of another object, provides a hook to
+/// observe when that object goes out of scope.
+public final class Lifetime {
+	/// A signal that sends a Completed event when the lifetime ends.
+	public let ended: Signal<(), NoError>
+
+	private let endedObserver: Observer<(), NoError>
```

Comment:
Can you use `Signal<(), NoError>.Observer`? That way it's clearer to see the parallel between the signal and its observer :)